catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/Listener.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * Responds to events from a {@link Harvester}. <b>{@code Listener}s are idempotent, receiving the
```

Comment:
✏️ Suggestion: _`Listener` implementations must be idempotent, i.e., receiving the same create, update or delete event multiple times for the same `HarvestedResource` must produce the same result_